docs/add-a-system-config.rst

```diff
@@ -74,373 +84,460 @@ where:
 
     MICROARCHITECTURE = CPU Microarchitecture
 
-    GPU = GPU Product Name
+    ACCELERATOR = ACCELERATOR Product Name
 
     NETWORK = Network Product Name
 
-In the `systems/all_hardware_descriptions/system_name` directory, add a
-`hardware_description.yaml` which follows the yaml format of existing
-`hardware_description.yaml` files.
+In the ``systems/all_hardware_descriptions/system_name`` directory, add a
+``hardware_description.yaml`` which follows the yaml format of existing
+``hardware_description.yaml`` files.
+
+.. _system-specification:
 
-2. Adding or Parameterizing System Software Stack
+B. Creating the System Definition (``system.py``)
 -------------------------------------------------
 
-``system.py`` in Benchpark provides an API to represent a system software stack as a
-command line parameterizable object. If none of the available software stack
-specifications match your system, you may add a `new-system` directory in the `systems`
-directory where the `new-system` directory name follows the naming convention:
+Now that you have defined the hardware description for your system, you can now create
+the ``system.py``, which involves defining the software on your system. This includes
+defining compilers and pre-installed packages, which your package manager can use
+instead of attempting to build the package from scratch. If using Spack, defining as
+many external packages as possible here will ensure a much faster build process, and
+using system-installed packages will likely always be more performant than building them
+from scratch.
+
+1. Creating the System class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we will recreate a fully-functional example of the AWS ``system.py``
+that we use for benchpark tutorials (see `aws-tutorial/system.py
+<https://github.com/LLNL/benchpark/blob/develop/systems/aws-tutorial/system.py>`_). To
+start, we import the base benchpark ``System`` class, which our ``AwsTutorial`` system
+will inherit from. We also import the maintainer and variant directives, which provide
+the utilities to track a maintainer by their GitHub username and variants to specify
+configurable properties of our system. We can specify the different AWS instances that
+share this same hardware and software specificaiton using the ``instance_type`` variant.
+We use ``instance_type`` here instead of ``cluster`` (you will see ``cluster`` in other
+systems), to differentiate a cloud ``instance`` from an HPC ``cluster``.
```

Comment:
How do we want to explain why we use ``instance_type`` over ``cluster`` for cloud?